Make the parser accept index list expressions

volmedo opened the pull request:
Description
-------------

Interfaces defined as instantiation of generic interfaces with more than one type parameter ([example](https://github.com/volmedo/mockery/blob/267dbdfe67d79c9a46cd76513734ec5e71ce80f4/pkg/fixtures/index_list_expression.go#L7)) are described as `*ast.IndexListExpr` by Go's AST library.

These interfaces were not accepted by the parser. This PR solves the issue by adding index list expressions as an accepted expression type.

- Fixes #856 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Version of Go used when building/testing:
---------------------------------------------

- [ ] 1.22
- [x] 1.23

How Has This Been Tested?
---------------------------

Added a new fixture illustrating the case and ran custom builds against it before and after the change, confirming the mock was produced as expected after the fix.

I also added a generator test for this specific case.

Checklist
-----------

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] (N/A) I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

LandonTClipp reviewed (changes requested):
Thanks, this looks mostly good. Can you just change the tests to follow the pattern here: https://github.com/vektra/mockery/tree/master/pkg/fixtures/type_alias

Basically:

1. Create subpackage in fixtures directory
2. Create interface
3. [Change `.mockery.yaml`](https://github.com/vektra/mockery/blob/master/.mockery.yaml#L97) to place mocks in that directory.
4. Add tests in that directory.

This is the model I'm moving towards. If you can do that I'll merge. Thanks!

volmedo commented:
I moved the fixtures and mocks to a subpackage and added some tests there as per the instructions. Let me know if I missed something.

On a side note, having actual tests in a folder called "fixtures" feels a bit weird. I'd create a new "tests" folder and move subpackages with test code in the "fixtures" folder there. Just a nitpick, feel free to disregard 🙂.

LandonTClipp reviewed (approved):
Looks great, thanks! Yeah the naming is hard, there is a lot of historical cruft in this repo that I haven't been able to clean up. Maybe soon.
